shared/output/ads/googleads/lib/google/ads/google_ads/v1/services/campaign_service/client.rb

```diff
@@ -120,7 +120,11 @@ def initialize
 
                 # Create credentials
                 credentials = @config.credentials
-                credentials ||= Credentials.default scope: @config.scope
+                enable_self_signed_jwt = @config.scope == Client.configure.scope &&
+                                         @config.endpoint == Client.configure.endpoint &&
+                                         !@config.endpoint.split(".").first.include?("-")
```

Comment:
Checking for the presence of the dash should be OK, NodeJS is doing something similar (they just strip the regional prefix and continue on their way).

Would there be any way to compare the `@config.endpoint` to a service name or some value which would be "documentai.googleapis.com"? If so, you could compare those values similarly to how you compare it against `Client.configure.endpoint`.

If that's not possible then this is LGTM with the note that by EOQ1 this bug _should_ be fixed and we can remove this logic. 

It's important to remember the worst effect this would have is to skip using self-signed JWT in a case where it shouldn't... which is what all the APIs are currently doing anyway.

Reply:
Unfortunately, we don't have a way to get the string `documentai.googleapis.com`. Only the regionized endpoint is set in the service configuration.